File: 1929-concatenation-of-array/1929-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int newN = n * 2;

        int[] ans = new int[newN];

        for(int i = 0; i<nums.length; i++){
            ans[i] = nums[i];
        }
        // for(int num : ans){
        //     System.out.print(num + " ");
        // }

        int counter = 0;
        for(int i = nums.length; i<newN; i++){
            ans[i] = ans[counter];
            counter++;
        }

        return ans;


    }
}

File: 1929-concatenation-of-array/1929-concatenation-of-array.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {

        let mut ans: Vec<i32> = Vec::new();

        for i in 0..nums.len()*2 {
            
            ans.push(nums[(i%nums.len())]);
        }

        return ans;
    }
}